### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by kryvets-vadym**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,91 @@
...
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side of the triangle must be greater than zero');
+    }
+
+    const maxSideLength = Math.max(a, b, c);
+
+    if (a + b + c - maxSideLength <= maxSideLength) {
```
Comment (taraselnik) [inline src/figuresClasses.ts:35]: cool

**Review by taraselnik — APPROVED** [review]
enjoy

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,91 @@
-export interface Figure {
+export enum Shape {
+  TriangleType = 'triangle',
+  CircleType = 'circle',
+  RectangleType = 'rectangle',
```
Comment (mykola-koval) [inline src/figuresClasses.ts:4]: why do you need `Type` everywhere?
```suggestion
  Triangle = 'triangle',
  Circle = 'circle',
  Rectangle = 'rectangle',
```

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,91 @@
...
+        + ' greater than the sum of the others');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+    const square = Math.sqrt(semiPerimeter
+      * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b)
+      * (semiPerimeter - this.c));
```
Comment (mykola-koval) [inline src/figuresClasses.ts:46]: let's use object destructuring to avoid using `this` everywhere

**Review by mykola-koval — APPROVED** [review]

**Pull request closed without merging**